File: Domain/Note.cs
using System;

namespace Domain
{
    public class Note
    {
        public Guid Id { get; set; }

        public String Name { get; set; }

        public String ProgressRating { get; set; }

        public String ExtraNote { get; set; }

        public DateTime DateAdded { get; set; }

        public Student Student { get; set; }

        public Guid StudentId { get; set; }

    }
}

File: client-app/src/app/models/note.ts
export interface INote {
  id: string;
  name: string;
  progressRating: string;
  extraNote: string;
  dateAdded: Date | null;
  studentId: string;
}

export class NoteFormValues implements INote {
  id: string = '';
  name: string = '';
  progressRating: string = '';
  extraNote: string = '';
  dateAdded: Date | null = new Date();
  studentId: string = '';

  constructor(init?: INote) {
    Object.assign(this, init);
  }
}
